File: Python3/1379. Find a Corresponding Node of a Binary Tree in a Clone of That Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def getTargetCopy(self, original: TreeNode, cloned: TreeNode, target: TreeNode) -> TreeNode:
        
        q = deque([(original, cloned)])
        
        while q:
            ori, clo = q.popleft()
            if ori == target:
                return clo
            
            if ori.left:
                q.append((ori.left, clo.left))
                
            if clo.right:
                q.append((ori.right, clo.right))
        
        
        

File: Java/1379. Find a Corresponding Node of a Binary Tree in a Clone of That Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */

class Solution {
    public final TreeNode getTargetCopy(final TreeNode original, final TreeNode cloned, final TreeNode target) {
        Queue<Pair<TreeNode, TreeNode>> q = new LinkedList<>();
        q.offer(new Pair<>(original, cloned));
        TreeNode ori, clo;
        
        while (!q.isEmpty()) {
            Pair<TreeNode, TreeNode> pair = q.poll();
            ori = pair.getKey();
            clo = pair.getValue();
            
            if (ori == target) {
                return clo;
            }
            
            if (ori.left != null) {
                q.offer(new Pair<>(ori.left, clo.left));
            }
            
            if (ori.right != null) {
                q.offer(new Pair<>(ori.right, clo.right));
            }
        }
        
        return null;
    }
}